base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1430,44 +1478,31 @@ where
                 )
                 .await?;
                 Ok(TransactionServiceResponse::SendMultisigUtxo(tx_id))
-            },
+            }.await,
         };
 
-        // If the individual handlers did not already send the API response then do it here.
-        if let Some(rp) = reply_channel {
-            let _result = rp.send(response).inspect_err(|_| {
-                warn!(target: LOG_TARGET, "Failed to send reply");
+        let _result = reply_channel
+            .send(response.inspect_err(|e1| warn!(target: LOG_TARGET, "{}", e1)))
+            .inspect_err(|e2| {
+                warn!(target: LOG_TARGET, "Failed to send reply: {:?}", e2);
             });
-        }
+
         Ok(())
     }
 
-    fn handle_get_fee_per_gram_stats_per_block_request(
+    async fn handle_get_fee_per_gram_stats_per_block_request(
         &self,
         count: u64,
-        reply_channel: oneshot::Sender<Result<TransactionServiceResponse, TransactionServiceError>>,
-    ) {
+    ) -> Result<FeePerGramStat, TransactionServiceError> {
         let connectivity = self.resources.connectivity.clone();
 
-        let query_base_node_fut = async move {
```

Comment:
this is bad, this call is now blocking, we cant block the entire wallet handler while we are busy dialing the base node